lib/spack/llnl/util/filesystem.py

```diff
@@ -84,26 +74,49 @@
 ]
 
 
-def getuid():
-    if is_windows:
-        import ctypes
+if sys.platform == "win32":
```

Comment:
The assumption here is that we'll import different implementations of the same functions in both conditional branches.